src/flat_map.rs

```diff
@@ -65,10 +60,12 @@ pub struct Values<'a, K: 'a, V: 'a> {
 }
 
 impl<K, V> FlatMap<K, V> {
+    #[inline]
```

Comment:
I went looking at the std lib for its usage of #[inline] on collections. For example, https://github.com/rust-lang/rust/blob/master/src/liballoc/btree/map.rs only has 5 usage of inline attribute. On the other hand, https://github.com/rust-lang/rust/blob/master/src/liballoc/vec.rs which is closer to what we're doing here has 50.

I went looking for the cost of using inline and it looks like it can slow compile times and increase binary sizes (maybe that doesn't matter here because this is generic code). Another option instead of aggressively using inline in a library is letting the application author use LTO: https://internals.rust-lang.org/t/when-should-i-use-inline/598

Just want to make sure you're aware of the costs before we decide this is the right thing.